TestGraphGenerator/Models/Graph.py

```diff
@@ -0,0 +1,13 @@
+from typing import List
+from dataclasses import dataclass, field
+
+from TestGraphGenerator.Models import Node, Relationship
+
+
+@dataclass
+class Graph:
```

Comment:
I'm not sure this should be a pure data class.
I would model it to have methods to add connections, that way you can add validation that the relationships are between existing nodes, that no relationship is added twice, etc.

Reply:
this is already happening in the graph engine so rechecking it will be a waste of resources